providers/base/metrics/provider.go

```diff
@@ -17,15 +17,22 @@ const (
 	ProviderLabel = "provider"
 	// IDLabel is a label for the ID of a provider response.
 	IDLabel = "id"
+	// ProviderTypeLabel is a label for the type of provider (WS, API, etc.)
+	ProviderTypeLabel = "type"
 	// StatusLabel is a label for the status of a provider response.
 	StatusLabel = "status"
 )
 
-type Status string
+type (
+	Status       string
+	ProviderType string
+)
 
 const (
-	Success Status = "success"
-	Failure Status = "failure"
+	Success    Status       = "success"
+	Failure    Status       = "failure"
+	WebSockets ProviderType = "websockets"
+	API        ProviderType = "api"
```

Comment:
see other comment, imo i think it makes more sense to move this to the provider types package